test/neighborsearch.jl

```diff
@@ -0,0 +1,59 @@
+@testset "Neighborsearch" begin
+  @testset "KBallSearcher" begin
+    # KBallSearcher
+
+    #create a regular grid to test
+    pdomain = RegularGrid(10,10)
```

Comment:
I would call this `D` or `dom` or something like it. The `pdomain` jargon was introduced in some solvers just to mean "problem domain". Here it doesn't feel very natural.